bundles/org.openhab.binding.surepetcare/src/main/resources/OH-INF/thing/things.xml

```diff
@@ -0,0 +1,619 @@
+
+	<channel-type id="photoType">
+		<item-type>Image</item-type>
+		<label>Pet Photo</label>
+		<description>The image of the pet</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="tagIdentifierType">
+		<item-type>String</item-type>
+		<label>Micro Chip Tag Identifier</label>
+		<description>The unique identifier of the pet micro chip tag</description>
+		<state readOnly="true" pattern="%s"/>
+	</channel-type>
+
+	<channel-type id="locationType">
+		<item-type>String</item-type>
+		<label>Location</label>
+		<description>The current location of the pet</description>
+		<state readOnly="false">
+			<options>
+				<option value="0">Unknown</option>
+				<option value="1">Inside</option>
+				<option value="2">Outside</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="locationChangedType">
+		<item-type>DateTime</item-type>
+		<label>Last Location Change Time</label>
+		<description>The time when the pet location has last changed</description>
+		<state readOnly="true" pattern="%1$tF %1$tR"/>
+	</channel-type>
+
+	<channel-type id="locationTimeoffsetType">
+		<item-type>String</item-type>
+		<label>Location Update with Time Offset</label>
+		<description>The location update of the pet with time offset (10, 30, 60 minutes ago.)</description>
+		<state readOnly="false">
+			<options>
+				<option value="10">10 mins ago</option>
+				<option value="30">30 mins ago</option>
+				<option value="60">60 mins ago</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="locationChangedThroughType">
+		<item-type>String</item-type>
+		<label>Pet Location Changed Through</label>
+		<description>Shows the device name or username of the last location change.</description>
+		<state readOnly="true" pattern="%s"/>
+	</channel-type>
+
+	<channel-type id="ledModeType" advanced="true">
+		<item-type>String</item-type>
+		<label>LED Mode</label>
+		<description>The current LED mode of the device</description>
+		<state readOnly="false">
+			<options>
+				<option value="0">Off</option>
+				<option value="1">Bright</option>
+				<option value="4">Dimmed</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="pairingModeType" advanced="true">
+		<item-type>String</item-type>
+		<label>Pairing Mode</label>
+		<description>The current pairing mode of the device</description>
+		<state readOnly="true">
+			<options>
+				<option value="0">Normal Mode</option>
+				<option value="1">Pairing Mode</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="onlineType">
+		<item-type>Switch</item-type>
+		<label>Online</label>
+		<description>Indicator if device is online or offline</description>
+		<state readOnly="true" pattern="%s"/>
+	</channel-type>
+
+	<channel-type id="curfewEnabledType">
+		<item-type>Switch</item-type>
+		<label>Curfew Enabled</label>
+		<description>Indicator if the curfew is enabled</description>
+		<state readOnly="false" pattern="%s"/>
```

Comment:
You don't need to specify patterns for Switch items.

Reply:
done